src/tools/rbd_mirror/Replayer.cc

```diff
@@ -220,6 +224,115 @@ int Replayer::init()
   return 0;
 }
 
+void Replayer::read_local_cluster_mirroring_info() {
```

Comment:
Alternative: on start-up, you could populate a `map<pool_id, set<global image id> >` for all local images.  On the first call to `set_sources`, remove from the collection all global image ids from your local collection that exist in the remote.  Any images that remain in the collection are local images that no longer exist on the mirror.